### ISSUE TITLE
Groups not functioning without sigh_mixin

### ISSUE THREAD

**Issue by jdswebb** [skypjack/entt#1145]
I've been trying to experiment with not using signals, but found when not using the sigh_mixin groups stop working. I'm not sure if this is a bug or intentional, but to reproduce take the example:

https://github.com/skypjack/entt/blob/master/test/example/signal_less.cpp

And add the line: `auto grp = registry.group<int>();`

Observe failure to compile due to `group_handler` implementation assuming `on_construct` exists.

Thanks

**Comment by skypjack** [comment]
Indeed, they don't. Confirmed. They were designed on top of the signal mechanism initially and I never changed it.
In theory, we can redesign them as a mixin today and avoid using signals. However, it's not super high prio at the moment.
